src/Plugin/PasswordConstraint/MinimumCharacters.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\password_enhancements\Plugin\PasswordConstraint;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\password_enhancements\Plugin\PasswordConstraintPluginInterface;
+
+/**
+ * Defines a common class for minimum character requirement.
+ */
+abstract class MinimumCharacters extends PluginBase implements PasswordConstraintPluginInterface {
+
+  /**
+   * The generated error message.
+   *
+   * @var string
+   */
+  protected $errorMessage;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->errorMessage = NULL;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getErrorMessage(): ?string {
+    return $this->errorMessage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInitialDescription(): string {
+    if ($this->configuration['minimum_characters'] > 1) {
+      return $this->configuration['descriptionPlural'];
+    }
+    else {
+      return $this->configuration['descriptionSingular'];
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(): array {
+    $form = [];
+
+    $form['minimum_characters'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Minimum characters'),
+      '#required' => TRUE,
+      '#min' => 1,
+      '#default_value' => !empty($this->configuration['minimum_characters']) ? $this->configuration['minimum_characters'] : 1,
```

Comment:
It can be null, that's why, I wanted to make sure that it gets the right values.
Also I'm checking it like this because it is more likely being set than not.

Reply:
> it is more likely being set than not.
Which means it could be upset, even if it is only 1% of times - and this is the edge case that you handle as simple and bulletproof as possible.

Reply:
So the basic ternary condition is `$condition ? true : false` the condition can be anything, but how I like to format the ternary conditions is like for the TRUE part I like to see the value that is being set, and for the FALSE part a default value only if the expected value is not available, yes I have to negate the condition, but I don't see any issue with that.